rpm/podman.spec

```diff
@@ -203,6 +207,14 @@ when `%{_bindir}/%{name}sh` is set as a login shell or set as os.Args[0].
 %autosetup -Sgit -n %{name}-%{version}
 sed -i 's;@@PODMAN@@\;$(BINDIR);@@PODMAN@@\;%{_bindir};' Makefile
 
+# These changes are only meant for copr builds
+%if %{defined copr_build}
+# podman --version should show short sha
+sed -i "s/^const RawVersion = .*/const RawVersion = \"##VERSION##-##SHORT_SHA##\"/" version/rawversion/version.go
+# use ParseTolerant to allow short sha in version
+sed -i "s/^var Version.*/var Version, err = semver.ParseTolerant(rawversion.RawVersion)/" version/version.go
```

Comment:
Minor suggestion: I had one of these "check the go source" version before, and it got annoying to fix every time something in go-land changed.  That's why I made the `test/version/version` Makefile target.  Run that, and your sed-ing can probably be swapped for a `printf` or an `echo`.

Reply:
clarification:

```
$ make test/version/version
$ test/version/version
4.7.0-dev
```

Reply:
the problem is `semver.MustParse` . `4.8.0-dev-SHORT_SHA` isn't strictly semver compliant. If I change the version in `version/rawversion/version.go` to `4.8.0-dev-foobar` . running `./test/version/version` will still give me `4.8.0-dev` and not `4.8.0-dev-foobar`.

Let me know if I misread and you meant something else.

Reply:
Oops, no, I read your `sed` command wrong.  This is **changing** `version.go` which seems wors-er (IMHO).  The issue is, `sed` will happily do nothing if no match is found.  It's also trivial for someone or some tool to modify `version.go` and break `sed` in a different or more silent way.

Perhaps it would be a slight improvement if you had a kind of "patch-template"?  Render that, verify it somehow, then apply it with `patch` (or the spec-file magic equivalent).  That way it might be more likely to fail-hard on a breaking code change?  I dunno, maybe there's a better/more golang or Makefile idiomatic way to override the version value in the code.

I bet @edsantiago or @vrothberg might have some ideas here, they're more inventive and  know the toolchain much better than I.

Reply:
i'm open to suggestions, but this is copr only, so not a big deal imho.

Reply:
That is true.  And the copr package ends up in a lot of places.  I don't think I'm smart enough to understand all the potential ramifications here.  Let's see what others say.